yt/fields/species_fields.py

```diff
@@ -157,22 +155,34 @@ def add_nuclei_density_fields(registry, ftype):
     for element in elements:
         registry.add_field((ftype, "%s_nuclei_density" % element),
                            sampling_type="local",
-                           function = _nuclei_density,
-                           units = unit_system["number_density"])
-    for element in ["H", "He"]:
+                           function=_nuclei_density,
+                           units=unit_system["number_density"])
+
+    for element in ["H", "He", "El"]:
         if element in elements:
             continue
         registry.add_field((ftype, "%s_nuclei_density" % element),
                            sampling_type="local",
-                           function = _default_nuclei_density,
-                           units = unit_system["number_density"])
+                           function=_default_nuclei_density,
+                           units=unit_system["number_density"])
+
 
 def _default_nuclei_density(field, data):
     ftype = field.name[0]
     element = field.name[1][:field.name[1].find("_")]
-    return data[ftype, "density"] * _primordial_mass_fraction[element] / \
-      ChemicalFormula(element).weight / data.ds.units.physical_constants.amu_cgs
-        
+    amu_cgs = data.ds.units.physical_constants.amu_cgs
+    if element == "El":
+        # This assumes full ionization!
```

Comment:
Could you use `default_mu` here?

Reply:
You can't because this is for the electron number density only and not the average number density (over all species).